sdk/provisioning/Azure.Provisioning/src/ProvisioningPlan.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Azure.Provisioning.Expressions;
+using Azure.Provisioning.Primitives;
+
+namespace Azure.Provisioning;
+
+// This is a draft to explore convenience APIs to make it easier to build tooling
+// around Azure infrastructure.  It's likely going to change a fair bit in the near future.
```

Comment:
There are a million other cool things we could do with a `ProvisioningPlan` like generating diagrams of your infrastructure, creating detailed reports of which identities have permissions to which resources, building a cost calculator that was tailored to your services/skus/regions/etc